### PR TITLE
Share get, watch resolve fn in pod controller

### PR DESCRIPTION
**Pull request by benjaminapetersen**
@spadgett @jwforres 
re todo #129

### CODE REVIEW

File: app/scripts/controllers/pod.js [hunk 1]
```diff
@@ -204,54 +205,47 @@ angular.module('openshiftConsole')
...
 
+    var podResolved = function(pod, action) {
+      $scope.loaded = true;
+      $scope.pod = pod;
+      var pods = {};
+      pods[pod.metadata.name] = pod;
+      setLogVars(pod);
+      setContainerVars();
+      ImageStreamResolver.fetchReferencedImageStreamImages(pods, $scope.imagesByDockerReference, $scope.imageStreamImageRefByDockerReference, requestContext);
+      $scope.containerTerminals = makeTerminals();
```
Comment (jwforres) [inline app/scripts/controllers/pod.js]: not sure this is a safe update, you've changed it so its now always calling makeTerminals on every update of the pod

File: app/scripts/controllers/pod.js [hunk 2]
```diff
@@ -204,54 +205,47 @@ angular.module('openshiftConsole')
...
     };
 
+    var podResolved = function(pod, action) {
+      $scope.loaded = true;
+      $scope.pod = pod;
+      var pods = {};
+      pods[pod.metadata.name] = pod;
+      setLogVars(pod);
+      setContainerVars();
+      ImageStreamResolver.fetchReferencedImageStreamImages(pods, $scope.imagesByDockerReference, $scope.imageStreamImageRefByDockerReference, requestContext);
```
Comment (jwforres) [inline app/scripts/controllers/pod.js]: I don't want to call this on an update, the image isn't going to change

Comment (benjaminapetersen) [inline app/scripts/controllers/pod.js]: np, got this

File: app/scripts/controllers/pod.js (hunk 1 above)
Comment (benjaminapetersen) [inline app/scripts/controllers/pod.js]: good catch here as well, moved that out into just the `.get()`

File: app/scripts/controllers/pod.js [hunk 3]
```diff
@@ -204,54 +205,51 @@ angular.module('openshiftConsole')
...
+      setContainerVars();
+      if (action === "DELETED") {
+        $scope.alerts["deleted"] = {
+          type: "warning",
+          message: "This pod has been deleted."
+        };
+      }
+      var pods = {};
+      pods[pod.metadata.name] = pod;
+      return pods;
```
Comment (jwforres) [inline app/scripts/controllers/pod.js]: this is a little odd to me. there isnt really a reason to have this here when the object stuff is only being used for the image stream resolve

Comment (benjaminapetersen) [inline app/scripts/controllers/pod.js]: Agree, is a little strange since its only for a fn arg.

Comment (benjaminapetersen) [inline app/scripts/controllers/pod.js]: Updated this & put the `pods[pod.metadata.name] = pod;` back as it was originally. 
rebased

**Comment by jwforres** [comment]
[merge]

**Pull request merged**